### PR TITLE
fix(action): Add many validations for setvar

### PR DESCRIPTION
**Pull request by jptosso**
Setvar wasn't doing many validations, and it was returning panics. Now it only allows TX as we don't support any other mutable variable

### CODE REVIEW

File: internal/actions/setvar.go [hunk 1]
```diff
@@ -33,6 +34,14 @@ func (a *setvarFn) Init(_ rules.RuleMetadata, data string) error {
 	var err error
 	key, val, valOk := strings.Cut(data, "=")
 	colKey, colVal, colOk := strings.Cut(key, ".")
+	// Right not it only makes sense to allow setting TX
+	// key is also required
+	if strings.ToUpper(colKey) != "TX" {
+		return errors.New("Invalid collection for setvar: " + colKey)
```
Comment (jcchavezs) [inline internal/actions/setvar.go:40]:
```suggestion
		return errors.New("invalid arguments, expected collection TX")
```

File: internal/actions/setvar.go [hunk 2]
```diff
@@ -33,6 +34,14 @@ func (a *setvarFn) Init(_ rules.RuleMetadata, data string) error {
...
+	}
+	if strings.TrimSpace(colVal) == "" {
+		return errors.New("Invalid key for setvar: " + colKey)
```
Comment (jcchavezs) [inline internal/actions/setvar.go:43]:
```suggestion
		return errors.New("invalid arguments, expected syntax TX.{key}={value}")
```

File: internal/actions/setvar.go [hunk 3]
```diff
@@ -71,7 +80,10 @@ func (a *setvarFn) Type() rules.ActionType {
 }
 
 func (a *setvarFn) evaluateTxCollection(r rules.RuleMetadata, tx rules.TransactionState, key string, value string) {
-	col := (tx.Collection(a.collection)).(collection.Map)
+	var col collection.Map
+	if c, ok := tx.Collection(a.collection).(collection.Map); ok {
```
Comment (jcchavezs) [inline internal/actions/setvar.go:84]:
```suggestion
	if col, ok := tx.Collection(a.collection).(collection.Map); !ok {
	    return
	}
```

File: internal/actions/setvar_test.go [hunk 4]
```diff
@@ -3,7 +3,22 @@
...
 
-import "testing"
+import (
+	"testing"
+)
+
+type md struct {
+}
+
+func (m *md) ID() int {
```
Comment (jcchavezs) [inline internal/actions/setvar_test.go:13]: nit: no need to use pointer

File: internal/actions/setvar.go [hunk 5]
```diff
@@ -71,7 +80,10 @@ func (a *setvarFn) Type() rules.ActionType {
...
+		col = c
+	}
 	if col == nil {
 		// fmt.Println("Invalid Collection " + a.Collection) LOG error?
```
Comment (M4tteoP) [inline internal/actions/setvar.go:88]: What about printing a `tx.DebugLogger().Error()` before returning?

Comment (jcchavezs) [reply internal/actions/setvar.go:88]: +1

File: internal/actions/setvar_test.go [hunk 6]
```diff
@@ -12,4 +27,22 @@ func TestSetvarInit(t *testing.T) {
 			t.Error("expected error ErrMissingArguments")
 		}
 	})
+	t.Run("non-map variable", func(t *testing.T) {
```
Comment (anuraaga) [inline internal/actions/setvar_test.go:30]: Nice

**Review by anuraaga — APPROVED** [review]

**Comment by jptosso** [comment]
I force merged because of codecov issues

**Pull request merged**